ejb/src/main/java/org/wildfly/httpclient/ejb/RequestBuilder.java

```diff
@@ -0,0 +1,220 @@
+        this.requestType = requestType;
+        return this;
+    }
+
+    RequestBuilder setInvocationId(final String invocationId) {
+        this.invocationId = invocationId;
+        return this;
+    }
+
+    RequestBuilder setVersion(final int version) {
+        this.version = version;
+        return this;
+    }
+
+    RequestBuilder setCancelIfRunning(final boolean cancelIfRunning) {
+        this.cancelIfRunning = cancelIfRunning;
+        return this;
+    }
+
+    // helper methods
+
+    ClientRequest createRequest(final String prefix) {
+        final ClientRequest clientRequest = new ClientRequest();
+        setRequestMethod(clientRequest);
+        setRequestPath(clientRequest, prefix);
+        setRequestHeaders(clientRequest);
+        return clientRequest;
+    }
+
+    private void setRequestMethod(final ClientRequest request) {
+        request.setMethod(requestType.getMethod());
+    }
+
+    private void setRequestPath(final ClientRequest request, final String prefix) {
+        if (requestType == START_INVOCATION) request.setPath(getStartEjbInvocationRequestPath(prefix));
+        else if (requestType == CREATE_SESSION) request.setPath(getCreateSessionEjbRequestPath(prefix));
+        else if (requestType == DISCOVER) request.setPath(getDiscoverEjbRequestPath(prefix));
+        else if (requestType == CANCEL_INVOCATION) request.setPath(getCancelEjbInvocationRequestPath(prefix));
+        else throw new IllegalStateException();
+    }
+
+    private void setRequestHeaders(final ClientRequest request) {
+        final HeaderMap headers = request.getRequestHeaders();
+        if (requestType == START_INVOCATION) {
+            headers.add(ACCEPT, INVOCATION_ACCEPT + "," + EJB_EXCEPTION);
+            headers.put(CONTENT_TYPE, INVOCATION.toString());
+            if (invocationId != null) {
+                headers.put(INVOCATION_ID, invocationId);
+            }
+            if (compressRequest) {
+                headers.put(CONTENT_ENCODING, GZIP.toString());
+            }
+            if (compressResponse) {
+                headers.put(ACCEPT_ENCODING, GZIP.toString());
+            }
+            headers.put(TRANSFER_ENCODING, CHUNKED.toString());
+        } else if (requestType == CREATE_SESSION) {
+            headers.add(ACCEPT, EJB_EXCEPTION.toString());
+            headers.put(CONTENT_TYPE, SESSION_OPEN.toString());
+        } else if (requestType == DISCOVER) {
+            headers.add(ACCEPT, EJB_DISCOVERY_RESPONSE + "," + EJB_EXCEPTION);
+        } else if (requestType != CANCEL_INVOCATION) {
+            throw new IllegalStateException();
+        }
+    }
+
+    private String getCreateSessionEjbRequestPath(final String prefix) {
+        final StringBuilder sb = new StringBuilder();
+        appendOperationPath(sb, prefix);
+        appendBeanPath(sb);
+        return sb.toString();
+    }
+
+    private String getDiscoverEjbRequestPath(final String prefix) {
```

Comment:
Comment indicating the format of the request path that this method builds?

Reply:
I typically add Javadoc comments only to public classes and methods, rather than to package-private or private ones.
Please note that this PR is part of ongoing refactoring, and some of these classes and methods may be removed later in the process.